python/kserve/kserve/protocol/grpc/servicer.py

```diff
@@ -34,6 +36,21 @@ def __init__(
         self._data_plane = data_plane
         self._mode_repository_extension = model_repository_extension
 
+    @classmethod
+    def validate_grpc_request(cls, request: pb.ModelInferRequest):
+        raw_inputs_length = len(request.raw_input_contents)
+        if raw_inputs_length != 0 and len(request.inputs) != raw_inputs_length:
+            raise InvalidInput(
+                f"the number of inputs ({len(request.inputs)}) does not match the expected number of "
+                f"raw input contents ({raw_inputs_length}) for model '{request.model_name}'."
+            )
+        if raw_inputs_length != 0:
+            for input_ in request.inputs:
+                if input_.HasField("contents"):
+                    raise InvalidInput(
+                        f"contents field must not be specified when using raw_input_contents for input '{input_.name}' for model '{request.model_name}'"
+                    )
```

Comment:
I think only FP16 and BF16 requires setting raw inputs, for other types user do not have to set raw input contents

Reply:
If raw_inputs_contents specified, then the contents field should be empty. https://github.com/kserve/open-inference-protocol/blob/52528cf4c48368114874e99217d25b9e006e494a/specification/protocol/open_inference_grpc.proto#L203-L204